components/ProjectPage/ProjectCard.tsx

```diff
@@ -0,0 +1,78 @@
+'use client';
+
+import { useEffect } from 'react';
+import { useAnimation } from 'framer-motion';
+import { useInView } from 'react-intersection-observer';
+import Image from 'next/image';
+import Link from 'next/link';
+
+import type { IProjectCard } from '@/types';
+import { useMediumQuery } from '@/hooks/useMediaQuery';
+
+import { MotionDiv } from '../Motion';
+
+const projectCardVariants = (index: number) => ({
+  visible: { opacity: 1, transition: { duration: 1 } },
+  hidden: {
+    opacity: 0,
+    transition: { duration: 1 },
+  },
+});
+
+const ProjectCard = ({
+  project,
+  index,
+}: {
+  project: IProjectCard;
+  index: number;
+}) => {
+  const [ref, isInView] = useInView({
+    threshold: 0.5,
+  });
+  const controls = useAnimation();
+  const isMedium = useMediumQuery();
+
+  useEffect(() => {
+    if (!isMedium) {
+      controls.start('visible');
+      return;
+    }
+    if (isInView) {
+      controls.start('visible');
+    } else {
+      controls.start('hidden');
+    }
+  }, [controls, isInView, isMedium]);
```

Comment:
I know we had discussed more complex animations that you wanted. You could get access to the exact threshold that each element is visible and update the animation from there. You can still do whatever movement you wanted.

You could add an animation that moves the element closer to the center:

```js
const projectCardVariants = (index: number) => ({
  visible: { opacity: 1, transition: { duration: 1 } },
  hidden: {
    opacity: 0,
    x:0,
    transition: { duration: 1 },
  },
  center:{
    opacity:1,
    x: index % 2 === 0 ? '100%' : '-100%',
    transition: { duration: 1 },
  },

});

```

Grab the `entry` from the useInview and add a second threshold:
```js
 const [ref, isInView,entry] = useInView({
    threshold: [0.25,0.65],
  });
```

And then rework the useEffect:
```js
  useEffect(() => {
    if (!isMedium) {
      controls.start('visible');
      return;
    }
    // if it's more than 65% visible, show the center animation
    if(entry?.intersectionRatio! > 0.65){
      controls.start('center');
      return;
    }
    if (isInView) {
      controls.start('visible');
    } else {
      controls.start('hidden');
    }
  }, [controls, isInView, isMedium,entry]);
  ```
  
  You can do this however you'd like. But I know you wanted more complex animations before.